src/pages/Facility/settings/locations/LocationForm.tsx

```diff
@@ -115,20 +126,58 @@ export default function LocationForm({
           pathParams: { facility_id: facilityId },
         }),
     onSuccess: () => {
+      toast.success(isEditMode ? t("location_updated") : t("location_created"));
+      queryClient.invalidateQueries({ queryKey: ["locations"] });
+
+      onSuccess?.();
+    },
+  });
+
+  const { mutate: submitBatch } = useMutation({
+    mutationFn: mutate(routes.batchRequest, { silent: true }),
+    onSuccess: (data: { results: BatchSubmissionResult[] }) => {
       toast.success(
-        location?.id ? t("location_updated") : t("location_created"),
+        t("bed_created_notification", { count: data.results.length }),
       );
       queryClient.invalidateQueries({ queryKey: ["locations"] });
-
       onSuccess?.();
     },
+
+    onError: () => {
+      toast.error(t("beds_creation_failed"));
+    },
   });
 
   function onSubmit(values: FormValues) {
+    if (values.form === "bd" && !isEditMode && (values.beds_count ?? 0) > 1) {
+      const data: LocationWrite = {
+        ...values,
+        mode: "instance",
+        description: values.description || "",
+        organizations: values.organizations,
+        parent: values.parent || undefined,
+      };
+
+      const batchRequest: BatchRequestBody = {
+        requests: Array.from(
+          { length: Number(values.beds_count) },
+          (_, index) => ({
+            url: `/api/v1/facility/${facilityId}/location/`,
+            method: "POST",
+            reference_id: `Location`,
```

Comment:
this would be better?

```suggestion
            reference_id: `location-${parentLocation's ID}`,
```

Reply:
what should be the id in case of adding in parent ?

Reply:
Done